src/NATS.Client.JetStream/NatsJSContext.Consumers.cs

```diff
@@ -125,4 +143,63 @@ public async ValueTask<bool> DeleteConsumerAsync(string stream, string consumer,
             cancellationToken);
         return response.Success;
     }
+
+    internal ValueTask<ConsumerInfo> CreateOrderedConsumerInternalAsync(
+        string stream,
+        NatsJSOrderedConsumerOpts opts,
+        CancellationToken cancellationToken)
+    {
+        var request = new ConsumerCreateRequest
+        {
+            StreamName = stream,
+            Config = new ConsumerConfiguration
+            {
+                Name = NuidWriter.NewNuid(),
+                DeliverPolicy = opts.DeliverPolicy,
+                AckPolicy = ConsumerConfigurationAckPolicy.none,
+                ReplayPolicy = opts.ReplayPolicy,
+                InactiveThreshold = opts.InactiveThreshold.ToNanos(),
+                NumReplicas = 1,
+                MemStorage = true,
+            },
+        };
+
+        if (opts.OptStartSeq > 0)
+        {
+            request.Config.OptStartSeq = opts.OptStartSeq;
+        }
+
+        if (opts.OptStartTime != default)
+        {
+            request.Config.OptStartTime = opts.OptStartTime;
+        }
+
+        if (opts.HeadersOnly)
+        {
+            request.Config.HeadersOnly = true;
+        }
+
+        if (opts.FilterSubjects.Length > 0)
+        {
+            request.Config.FilterSubjects = opts.FilterSubjects;
+        }
+
+        var subject = $"{Opts.Prefix}.CONSUMER.CREATE.{request.StreamName}";
+
+        if (!string.IsNullOrWhiteSpace(request.Config.Name))
+        {
+            subject += $".{request.Config.Name}";
+            request.Config.Name = default!;
+        }
+
+        if (!string.IsNullOrWhiteSpace(request.Config.FilterSubject))
+        {
+            subject += $".{request.Config.FilterSubject}";
+        }
+
```

Comment:
What if the name was not set, but the filter subject was set?  With this logic the subject token for name is not present, and the filter subject token is appended to where the name subject should go